### PR TITLE
SNOW-623421 Fixes DisableTelemetry to disable telemetry when it is true

### LINKED ISSUE #613: DisableTelemetry is ignored
Just looking through the code it's quite obvious.

First there is a check that set a disabled telementry client:

https://github.com/snowflakedb/gosnowflake/blob/3c7253cb065fc33771a2e9b46c5ae8c00f321fdd/connection.go#L448-L450

However it's overwritten a couple of lines later:

https://github.com/snowflakedb/gosnowflake/blob/3c7253cb065fc33771a2e9b46c5ae8c00f321fdd/connection.go#L477-L482

### PR DESCRIPTION
**Pull request by SimbaGithub**
### Description
Simba ticket 00390103
Fixes #613 where the connection parameter DisableTelemetry is ignored.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sfc-gh-jbahk — APPROVED** [review]

File: telemetry_test.go [hunk 1]
```diff
@@ -106,7 +106,10 @@ func TestDisableTelemetry(t *testing.T) {
 	if err = authenticateWithConfig(sc); err != nil {
 		t.Fatal(err)
 	}
-	if _, err = sc.Query("select 1", nil); err != nil {
-		t.Fatal(err)
+	if !sc.cfg.DisableTelemetry {
+		t.Errorf("DisableTelemetry should be true. DisableTelemetry: %v", sc.cfg.DisableTelemetry)
+	}
+	if sc.telemetry.enabled {
+		t.Errorf("telemetry should be disabled.")
```
Comment (sfc-gh-hchaturvedi) [inline telemetry_test.go:113]: Can you also add a test that it is enabled when it is set to true? Also what's the default value?

File: connection.go [hunk 2]
```diff
@@ -478,7 +475,7 @@ func buildSnowflakeConn(ctx context.Context, config Config) (*snowflakeConn, err
 		flushSize: defaultFlushSize,
 		sr:        sc.rest,
```
[the remark this answers is not in the document]
Comment (SimbaGithub) [reply connection.go]: You're right. Thanks for pointing it out. I've updated the code.

File: telemetry_test.go (hunk 1 above)
Comment (SimbaGithub) [reply telemetry_test.go:113]: The test case for telemetry enabled has been added.
The default value of DisableTelemetry is false, i.e. telemetry is always enabled.

**Review by Thomasdezeeuw — APPROVED** [review]
Thanks @SimbaGithub

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
LGTM

**Pull request merged**